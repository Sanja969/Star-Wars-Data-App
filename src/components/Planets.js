const Planets = () => {
  const text = 'Planets';
  return (
    <p>{text}</p>
  );
};

export default Planets;
