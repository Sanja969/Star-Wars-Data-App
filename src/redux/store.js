import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import charactersReducer from './characters';
import planetsReducer from './planets';
import filmsReducer from './films';
import shipsReducer from './starships';
import detailReducer from './detail';

const rootReducer = combineReducers({
  characters: charactersReducer,
  planets: planetsReducer,
  films: filmsReducer,
  ships: shipsReducer,
  detail: detailReducer,
});

const store = configureStore({
  reducer: rootReducer,
  middleware: [thunk, logger],
});

export default store;
