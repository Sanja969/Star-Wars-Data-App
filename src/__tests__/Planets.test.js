import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Planets from '../components/Planets';

describe('Test Planets render', () => {
  test('Test Snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <Planets />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
