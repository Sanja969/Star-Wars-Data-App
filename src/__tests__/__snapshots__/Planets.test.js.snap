// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Planets render Test Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-white pt-4 text-center"
      >
        <div
          class="ps-3 d-flex align-items-center"
        >
          <button
            class="h1"
            type="button"
          >
            ⇦
          </button>
          <h4
            class="ps-5 ms-5"
          >
            Planets
          </h4>
        </div>
        <input
          class="search w-75 text-center"
          placeholder="Search..."
          type="search"
        />
        <div
          class="d-flex flex-wrap w-100"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-white pt-4 text-center"
    >
      <div
        class="ps-3 d-flex align-items-center"
      >
        <button
          class="h1"
          type="button"
        >
          ⇦
        </button>
        <h4
          class="ps-5 ms-5"
        >
          Planets
        </h4>
      </div>
      <input
        class="search w-75 text-center"
        placeholder="Search..."
        type="search"
      />
      <div
        class="d-flex flex-wrap w-100"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
